docs/development.md

```diff
@@ -0,0 +1,250 @@
+difference is that it is not possible to run `pip install` inside Nix shell
+(because of the way Nix works) and it is necessary to use the built-in
+`setuptoolsShellHook` function instead.
+
+We also provide a special Nix environment for retrieving and preparing kernel
+versions necessary for running [regression tests](#python-tests)
+(`nix develop .#test-kernel-buildenv`).
+
+### Local development environment
+
+You can also develop DiffKemp directly. For this you need to install the
+necessary [dependencies](installation.md#dependencies) to your system and then
+[build DiffKemp](#build).
+
+### Docker
+
+> [!CAUTION]
+> The docker container is not currently maintained and it is recommended to use
+> [Nix flake](#nix) instead.
+
+We also provide development container image prepared that can be retrieved from
+DockerHub:
+[https://hub.docker.com/r/viktormalik/diffkemp-devel/](https://hub.docker.com/r/viktormalik/diffkemp-devel/)
+
+After that, the container can be run using
+
+```txt
+docker/diffkemp-devel/run-container.py [--llvm-version LLVM_VERSION]
+                                       [--build-dir BUILD_DIR]
+                                       [--diffkemp-dir DIFFKEMP_DIR]
+                                       [--image IMAGE]
+```
+
+The script mounts the current directory as a volume inside the container.
+Then it automatically builds SimpLL in `BUILD_DIR` (default is "build") using
+`LLVM_VERSION` (default is the newest supported version) and installs DiffKemp.
+
+If running multiple containers at the same time, you need to specify a unique
+`BUILD_DIR` for each one.
+
+If running the container from a different directory than the root DiffKemp
+directory, you need to specify where DiffKemp is located using the
+`--diffkemp-dir` option.
+
+By default, the DockerHub image is used, but a custom image may be set using
+the `--image` option.
+
+## Build
+
+To build DiffKemp, use the following commands:
+
+```sh
+cmake -S . -B build -GNinja -DBUILD_VIEWER=On
+ninja -C build
+pip install -e . # In case of Nix use setuptoolsShellHook instead
+```
+
+- `-DBUILD_VIEWER=On`: This flag will install packages and build the result
+  viewer. It will try to install the packages on every `cmake` run, so you may
+  want to use `-DBUILD_VIEWER=Off` instead.
+
+If you make changes to the SimpLL library, you will need to rebuild it by
+running `ninja -C build`. We are using [CFFI](https://cffi.readthedocs.io/en/stable/)
+for accessing the library from the Python. The CFFI is not rebuilt by default,
+so even if you rebuild the library, the changes will not be visible when running
+`diffkemp compare ...`. Currently, you have multiple options to overcome this:
+
+1. Use `diffkemp compare --disable-simpll-ffi ...` which will run SimpLL
+  through a binary instead of the CFFI.
+2. On some OS distributions, it is enough to run `pip install -e .` again.
+3. Another option is to to run `rm build/_simpll.abi3.so` and then run `cmake`
```

Comment:
I'd suggest replacing `rm build/_simpll.abi3.so` with `tools/simpll_clean.sh`

Reply:
Unfortunately, in this situation the script does not help -- it exits without doing anything because the version of `LLVM IR` and of `SimpLL` library would be the same:
https://github.com/diffkemp/diffkemp/blob/eb7511df05bd27347df3eac27eca7e6cc83713a1/tools/simpll_clean.sh#L32-L37

The script (`tools/simpll_clean.sh`) could be mentioned somewhere, but I would rather leave it after #344 is merged if the script is still useful (also the script will probably need to be updated in #344 or after that).